### PR TITLE
HPCC-2968 ConfigMgr should allow hostnames in computer configuration

### PR DESCRIPTION
**Pull request by rpastrana**
- Fixes new/new range processing to work as originally intended
- New computer process allows hostnames
- Adds new functions for IP/hostname parsing

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [x] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [x] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [x] I have tested my changes in multiple modern browers
  - [x] The component(s) render as expected

### CODE REVIEW

**Comment by rpastrana** [comment]
This change in the ConfigManager might impact many components including Roxy/Thor/DFU/DALI/DZESP

**Comment by richardkchapman** [comment]
@rpastrana Who should review this?

**Comment by rpastrana** [comment]
@wangkx please review.

File: esp/files/scripts/configmgr/navtree.js [hunk 1]
```diff
@@ -4367,12 +4396,15 @@ function ipCount(){
...
-  
+  const ipPattern =/^(\d{1,3})\.(\d{1,3})\.(\d{1,3})\.(\d{1,3})$/;
+  const ipRangePattern="([0-9][0-9]?[0-9]?)\.([0-9][0-9]?[0-9]?)\.([0-9][0-9]?[0-9]?)\.([0-9][0-9]?[0-9]?\-[0-9][0-9]?[0-9]?)"
+
   for(var k=0; k < ipArr.length ;k++ )
   {
-
-    if (ipArr[k].match("-") != null )
+	  var currentval = ipArr[k]; 
+	  var mymatch = currentval.match(ipRangePattern);
```
Comment (wangkx) [inline esp/files/scripts/configmgr/navtree.js]: Remove the last 2 lines.

**Review by wangkx — COMMENTED** [review]
Several minor comments and a question.

File: esp/services/WsDeploy/WsDeployService.cpp [hunk 2]
```diff
@@ -4722,21 +4722,58 @@ bool CWsDeployFileInfo::handleComputer(IEspContext &context, IEspHandleComputerR
...
+    StringBuffer attr, val, sAttributes;
+    attr.appendf(" %s=\"%s\" %s=\"%s\"", &XML_ATTR_DOMAIN[1], domain, &XML_ATTR_COMPUTERTYPE[1], cType);
 
-    resp.setCompName(sbNewName.str());
+    StringBuffer sNode("<" XML_TAG_HARDWARE ">");
+    StringBuffer strCheckXPath;
+    strCheckXPath.setf("%s/%s[%s=\"%s\"][1]", XML_TAG_HARDWARE, XML_TAG_COMPUTER, XML_ATTR_NETADDRESS, netAddress);
+
+    sNode.appendf("<" XML_TAG_COMPUTER " %s=\"%s\" %s=\"%s\" %s/>",
+                  &XML_ATTR_NAME[1], getUniqueName(pEnvRoot, sName, XML_TAG_COMPUTER, XML_TAG_HARDWARE),
```
Comment (rpastrana) [inline esp/services/WsDeploy/WsDeployService.cpp]: appends "_X" where X > 1 and name_X is unique

**Comment by rpastrana** [comment]
@wangkx please review latest changes.

**Comment by wangkx** [comment]
@rpastrana looks fine.

**Comment by rpastrana** [comment]
@richardkchapman ready to be merged.

**Comment by richardkchapman** [comment]
@rpastrana You checked the "Requires a change to the documentation" box but not either of "I have updated the documentation accordingly", or... "I have created a JIRA ticket to update the documentation." Do we need a Jira creating?

**Comment by rpastrana** [comment]
Doc Jira subtask: https://track.hpccsystems.com/browse/HPCC-17132
@richardkchapman

**Pull request merged**